spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/hazelcast/AbstractHazelcastAutoConfigurationTests.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.hazelcast;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+/**
+ * Base class for {@link HazelcastAutoConfiguration} tests.
+ *
+ * @author Stephane Nicoll
+ * @author Vedran Pavic
+ */
+public class AbstractHazelcastAutoConfigurationTests {
```

Comment:
Meh. I'd really like we avoid separating things like that. In particular we could test what happens when both the server and the client is available (the client wins?). And then we could add an explicit `spring.hazelcast.mode` that sets the other one and we can validate the right instance has been auto-configured.